src/index.ts

```diff
@@ -231,6 +233,70 @@ function launchMetals(
       );
     }
 
+    registerCommand("metals.new-scala-file", async () => {
+      const currentDoc = await workspace.document;
+      const currentPath = currentDoc.uri;
+      const parentDir = path.dirname(currentPath);
+
+      const fileOptions = [
+        {
+          kind: "class",
+          value: 0,
+          label: "Class"
+        },
+        {
+          kind: "object",
+          value: 1,
+          label: "Object"
+        },
+        {
+          kind: "trait",
+          value: 2,
+          label: "Trait"
+        },
+        {
+          kind: "package-object",
+          value: 3,
+          label: "Package Object"
+        },
+        {
+          kind: "worksheet",
+          value: 4,
+          label: "Worksheet"
+        }
+      ];
+
+      const fileSelection = await workspace.showQuickpick(
+        fileOptions.map(option => option.label),
+        "Select the kind of file to create"
+      );
+
+      const desiredFileType =
+        fileOptions.find(option => fileSelection === option.value)?.kind || "";
+
+      if (desiredFileType !== "") {
+        const fileName = await workspace.callAsync<string>("input", [
+          "Name:",
+          ""
+        ]);
+        const arg: MetalsNewScalaFileParams = {
+          directory: parentDir,
+          name: fileName,
+          kind: desiredFileType
+        };
+        client
+          .sendRequest(ExecuteCommandRequest.type, {
+            command: "new-scala-file",
+            arguments: [arg]
+          })
+          .then(result => {
```

Comment:
nit: it's a bit weird to mix and match `async/await` and `then/catch`. In metals-vscode I've opted for banning `async/await`, since it makes it harder to handle errors in a compositional way.

Just food for thought, not a blocker for this PR

Reply:
Good point, I shouldn't have mixed this. My main reason for making this `async/await` instead of `then/catch` was because in the vs code extension I really dislike the 3 levels of nested `then`. I find it a bit hard to read.

_However_, your point is a good one about composing errors. Let me think on this, and maybe I'll just follow suit, and remove all the `async/await` to have more "solidarity" with the vs code code.

Reply:
I did change this up though not to mix them.

Reply:
Yes, I agree the nested then are not great, but I’m under the impression the main issue is with the length of the body, which I aim to breaking down soon